action/protocol/staking/protocol.go

```diff
@@ -206,17 +209,58 @@ func (p *Protocol) CreatePreStates(ctx context.Context, sm protocol.StateManager
 	bcCtx := protocol.MustGetBlockchainCtx(ctx)
 	blkCtx := protocol.MustGetBlockCtx(ctx)
 	hu := config.NewHeightUpgrade(&bcCtx.Genesis)
-	if blkCtx.BlockHeight != hu.GreenlandBlockHeight() {
+	if blkCtx.BlockHeight == hu.GreenlandBlockHeight() {
+		csr, err := ConstructBaseView(sm)
+		if err != nil {
+			return err
+		}
+		_, err = sm.PutState(csr.BaseView().bucketPool.total, protocol.NamespaceOption(StakingNameSpace), protocol.KeyOption(bucketPoolAddrKey))
+		if err != nil {
+			return err
+		}
+	}
+
+	if p.candidatesBucketsIndexer == nil {
+		return nil
+	}
+	rp := rolldpos.MustGetProtocol(protocol.MustGetRegistry(ctx))
+	epochStartHeight := rp.GetEpochHeight(rp.GetEpochNum(blkCtx.BlockHeight))
+	if epochStartHeight != blkCtx.BlockHeight {
 		return nil
 	}
-	csr, err := ConstructBaseView(sm)
+	// only for after fairbank
+	if hu.IsPre(config.Fairbank, epochStartHeight) {
+		return nil
+	}
```

Comment:
don't need this. staking protocol is effective only after fairbank height

Reply:
deleted